src/async.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 Adobe
+// All Rights Reserved.
+//
+// NOTICE: Adobe permits you to use, modify, and distribute this file in accordance with the terms
+// of the Adobe license agreement accompanying it.
+
+// identity
+#include "orc/async.hpp"
+
+// stdc++
+#include <cassert>
+
+// application
+#include "orc/settings.hpp"
+#include "orc/task_system.hpp"
+#include "orc/tracy.hpp"
+
+/**************************************************************************************************/
+
+namespace orc {
+
+/**************************************************************************************************/
+
+namespace {
+
+/**************************************************************************************************/
+
+struct work_counter {
+    struct state {
```

Comment:
This is a 'struct' - did you mean `state` to be public? (And if yes, why have the `public` decl later on?)

Reply:
Yes - this is all supposed to be public to make access easier. Note that all this is within a single source file, so the public nature of these structs is encapsulated from the rest of the system. What's in here is all related to async work item management, so public access makes the code easier to grok. I've remove the extra `public` qualifier below.